src/app/students/(table)/student-table.tsx

```diff
@@ -38,10 +40,50 @@ const DataTable = <TData, TValue>({
     columns,
     data,
 }: DataTableProps<TData, TValue>) => {
-    const [rowSelection, setRowSelection] = React.useState({})
-    const [columnVisibility, setColumnVisibility] = React.useState<VisibilityState>({})
-    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([])
-    const [sorting, setSorting] = React.useState<SortingState>([])
+    const [rowSelection, setRowSelection] = useState({})
+    const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({})
+    const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+    const [sorting, setSorting] = useState<SortingState>([])
+
+    const [pagination, setPagination] = useState<PaginationState>({
+        pageIndex: 1,
```

Comment:
wait but if this is 1, why do we add 1 later below

Reply:
think this will be removed from my next upcoming pr (which was sort of a refactor so that pagination and searching is central and done through the useStudents hook/context)